src/facebook/facebook-notifications/facebooknotificationsyncadaptor.cpp

```diff
@@ -87,11 +87,17 @@ void FacebookNotificationSyncAdaptor::requestNotifications(int accountId, const
     queryItems.append(QPair<QString, QString>(QString(QLatin1String("locale")), QLocale::system().name()));
     QUrl url(graphAPI(QLatin1String("/me/notifications")));
     if (pagingToken.isEmpty()) {
-        int sinceSpan = m_accountSyncProfile
-                      ? m_accountSyncProfile->key(Buteo::KEY_SYNC_SINCE_DAYS_PAST, QStringLiteral("7")).toInt()
-                      : 7;
+        QDateTime since = lastSyncTimestamp(QLatin1String("facebook"),
+                                            SocialNetworkSyncAdaptor::dataTypeName(SocialNetworkSyncAdaptor::Notifications),
+                                            accountId);
+        if (!since.isValid()) {
+            int sinceSpan = m_accountSyncProfile
+                          ? m_accountSyncProfile->key(Buteo::KEY_SYNC_SINCE_DAYS_PAST, QStringLiteral("7")).toInt()
+                          : 7;
+            since = QDateTime::currentDateTime().addDays(-1 * sinceSpan);
+        }
         queryItems.append(QPair<QString, QString>(QString(QLatin1String("since")),
-                          QString::number(QDateTime::currentDateTime().addDays(-1 * sinceSpan).toTime_t())));
+                          QString::number(since.toTime_t())));
```

Comment:
is this timestamp utc?  ie, is it the same format which is "expected" by Facebook?

If the user sets different time/date settings on their device, I guess they might "miss" notifications from Facebook... I wonder if this is a problem we need to consider, or whether that behaviour is acceptable in that case.